propagation_test.go

```diff
@@ -179,31 +179,46 @@ func TestBaggagePropagationHTTP(t *testing.T) {
 }
 
 func TestJaegerBaggageHeader(t *testing.T) {
-	metricsFactory, metrics := initMetrics()
-	tracer, closer := NewTracer("DOOP",
-		NewConstSampler(true),
-		NewNullReporter(),
-		TracerOptions.Metrics(metrics),
-	)
-	defer closer.Close()
+	var testcases = []struct {
+		refFunc func(opentracing.SpanContext) opentracing.SpanReference
+	}{
+		{
+			refFunc: opentracing.ChildOf,
+		},
+		{
+			refFunc: opentracing.FollowsFrom,
+		},
+	}
 
-	h := http.Header{}
-	h.Add(JaegerBaggageHeader, "key1=value1, key 2=value two")
+	for _, testcase := range testcases {
 
-	ctx, err := tracer.Extract(opentracing.HTTPHeaders, opentracing.HTTPHeadersCarrier(h))
-	require.NoError(t, err)
+		metricsFactory, metrics := initMetrics()
+		tracer, closer := NewTracer("DOOP",
+			NewConstSampler(true),
+			NewNullReporter(),
+			TracerOptions.Metrics(metrics),
+		)
+		defer closer.Close()
```

Comment:
defer inside a loop is not great.  Better to do this
```go
for _, testcase := range testcases {
    t.Run("", func( t *testing.T) {
        // test code here
    })
```

Reply:
Certainly. Since this PR is already merged, I'll close the loop with a followup PR.

Reply:
Fixed in #315.